nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/src/main/nifi/src/app/pages/provenance/state/provenance-event-listing/provenance-event-listing.effects.ts

```diff
@@ -311,18 +321,31 @@ export class ProvenanceEventListingEffects {
                             dialogReference.componentInstance.replay
                                 .pipe(takeUntil(dialogReference.afterClosed()))
                                 .subscribe(() => {
-                                    this.provenanceService.replay(request.id).subscribe(() => {
-                                        this.store.dispatch(
-                                            ProvenanceEventListingActions.showOkDialog({
-                                                title: 'Provenance',
-                                                message: 'Successfully submitted replay request.'
-                                            })
-                                        );
+                                    this.provenanceService.replay(request.id).subscribe({
+                                        next: () => {
+                                            dialogReference.close();
+
+                                            this.store.dispatch(
+                                                ProvenanceEventListingActions.showOkDialog({
+                                                    title: 'Provenance',
+                                                    message: 'Successfully submitted replay request.'
+                                                })
+                                            );
+                                        },
+                                        error: (errorResponse: HttpErrorResponse) => {
+                                            this.store.dispatch(
+                                                ErrorActions.snackBarError({ error: errorResponse.error })
+                                            );
```

Comment:
A snackbar in this case is a bit odd. The dialog is open and the snackbar pops up well under the dialog. almost making it seem unrelated to the action they just attempted (replay of the event). Maybe a banner here?

<img width="795" alt="Screenshot 2024-02-13 at 1 31 41 PM" src="https://github.com/apache/nifi/assets/713866/795d2b2f-f389-4872-a596-3794fe3b9f73">

Reply:
I made this choice because the error isn't based on user input (there's nothing they can change and resubmit) in the dialog and a snackbar error was used in similar situations like when loading descriptors for new services and parameters.

Reply:
I will also update to close the dialog so it error is shown more clearly.